junit-jupiter-params/src/main/java/org/junit/jupiter/params/aggregator/ArgumentsAccessor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.params.aggregator;
+
+import java.util.List;
+
+public interface ArgumentsAccessor {
+
+	Object[] toArray();
+
+	List<Object> toList();
+
+	int size();
+
+	Object get(int index);
+
+	<T> T get(Class<T> clazz, int index);
```

Comment:
Please swap the order of the formal parameters: the required type is an optional argument and should therefore come after `index`.

Reply:
Please rename `clazz` to `requireType`.